jOOQ/src/main/java/org/jooq/tools/Convert.java

```diff
@@ -517,7 +522,18 @@ else if (Collection.class.isAssignableFrom(toClass) &&
                 else if (toClass.isArray()
                         && Collection.class.isAssignableFrom(fromClass)){
                     Collection f = (Collection) from;
-                    return (U) f.toArray();
+                    Class componentType = toClass.getComponentType();
+
+                    Object[] dest = (Object[]) Array.newInstance(componentType, f.size());
+                    Object[] list = f.stream()
+                            .map(e -> {
+                                if (!componentType.isAssignableFrom(e.getClass()))
+                                    return convert(e, componentType);
+                                return e;
+                            }).toArray();
+                    System.arraycopy(list, 0, dest, 0, dest.length);
+
+                    return (U) dest;
                 }
```

Comment:
We can't generally make use of the `Stream` API in jOOQ as we still support Java 6. I will fix this code to retrofit it to our expectations.